apps/api/src/billing/services/wallet-initializer/wallet-initializer.service.ts

```diff
@@ -28,6 +29,9 @@ export class WalletInitializerService {
         },
         { returning: true }
       );
+    } catch (error) {
+      await this.userWalletRepository.deleteById(userWallet.id);
```

Comment:
thought (blocking): I think we should revoke the authz as well here

Reply:
regarding revoking, we don't know the reason why it failed, very likely because trial spending was not granted. I haven't dig deep into implementation details of `createAndAuthorizeTrialSpending`. It can potentially fail on `updateById` which is less likely then yes, we need to revoke trial spending

Reply:
Yea I agree the `updateById` has a very low chance of failing

Reply:
On the other hand, if we drop the wallet the pk is already incremented. It will never be used again. The grants will expire in a month on their own.

Reply:
Good point